compiler/semcall.nim

```diff
@@ -235,24 +236,27 @@ proc presentFailedCandidates(c: PContext, n: PNode, errors: CandidateErrors):
         candidates.add "\n  but expression '"
         if err.firstMismatch.kind == kVarNeeded:
           candidates.add renderNotLValue(nArg)
+          immutParam = renderNotLValue(nArg)
```

Comment:
pre-existing but now that you use `quoteExpr` how about avoiding un-matched single quotes as much as possible so code is correct "by construction" (and also deals with nested quotes in a correct way, see my other comment regarding escaping quotes):
```nim
candidates.add "\n  but expression "
if err.firstMismatch.kind == kVarNeeded:
  immutParam = renderNotLValue(nArg)
  candidates.add "$1 is immutable, not 'var'" % renderNotLValue(nArg).quoteExpr
else:
  candidates.add "$1 is of type: " % renderTree(nArg).quoteExpr
```